xarray/tests/test_indexing.py

```diff
@@ -622,3 +622,8 @@ def test_create_mask_error():
 def test_posify_mask_subindexer(indices, expected):
     actual = indexing._posify_mask_subindexer(indices)
     np.testing.assert_array_equal(expected, actual)
+
+
+def test_indexing_1d_object_array():
```

Comment:
I think (one voice of many) that if mark failing tests that we'd like to fix as `xfail` we should merge them